### PR TITLE
feat: add codemod for sc and emotion

### LINKED ISSUE #179: Code mod from css prop
We have an internal code mod that we used from styled components to emotion. Let's re-use that.

### LINKED ISSUE #239: Code mod from styled component api

### PR DESCRIPTION
**Pull request by pgmanutd**
This closes #239 
This closes #179

### CODE REVIEW

File: packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+    const Component = (props) => (
+      <>
+        <div
+          css={\`
+            color: red;
+            background-color: #000;
+          \`}
+        />
+        <span css={\` color: blue; \`} />
```
Comment (madou) [inline packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx:103]: we have a issue here #313 to add `css` to improve syntax highlighting (and also enable mixins) - we will need to come bask and update the codemod when we implement it - or alternatively - should we not codemod it for now? 🤔

File: packages/core/src/codemods/emotion-to-compiled/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# emotion-to-compiled
+
+> Codemod for easy migration from emotion.
+
+```bash
+# https://github.com/facebook/jscodeshift#usage-cli
+npx jscodeshift --parser=tsx --extensions=tsx --transform node_modules/@compiled/core/dist/codemods/emotion-to-compiled src
```
Comment (madou) [inline packages/core/src/codemods/emotion-to-compiled/README.md:7]: nit: should we be explicit around `src` here and where devs should point to?

File: packages/core/src/codemods/styled-components-to-compiled/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# styled-components-to-compiled
+
+> Codemod for easy migration from styled components.
+
+```bash
+# https://github.com/facebook/jscodeshift#usage-cli
+npx jscodeshift --parser=tsx --extensions=tsx --transform node_modules/@compiled/core/dist/codemods/styled-components-to-compiled src
```
Comment (madou) [inline packages/core/src/codemods/styled-components-to-compiled/README.md:7]: same nit here for `src`

File: yarn.lock [hunk 4]
```diff
@@ -46,6 +46,28 @@
     semver "^5.4.1"
     source-map "^0.5.0"
 
+"@babel/core@^7.1.6":
```
Comment (madou) [inline yarn.lock:49]: can you de-dupe this please

`npx yarn de-duplicate`

**Comment by madou** [comment]
Really good! Tiny nits but its good to go pretty much!

Want to run it on some AK package to verify it works?

File: packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx (hunk 1 above)
Comment (pgmanutd) [reply packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx:103]: 😅 Lets do it for this release. I ll leave a comment on #313 or probably create a card? What you think? Its just matter of removing the code.

Comment (madou) [reply packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx:103]: sounds good

we might need to add a codemod to add it back too hehe

Comment (pgmanutd) [reply packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx:103]: 🤔 hehe yeah. Atleast we will have first release 🎉

Comment (pgmanutd) [reply packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx:103]: https://github.com/atlassian-labs/compiled/issues/340

File: packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx [hunk 5]
```diff
@@ -283,7 +283,75 @@ describe('emotion-to-compiled transformer', () => {
...
+            Hello
+          </SomeComponent>
+        )}
+      </ClassNames>
+    );
+    `,
+    `
+    /* TODO: (from codemod) Emotion's JSX pragma has been removed. Please import appropriate JSX transformer.
+    Eg. import React from 'react'; */
+    /* TODO: (from codemod) "ClassNames" is not exported from "@compiled/core" at the moment. Please find an alternative for it. */
```
Comment (madou) [inline packages/core/src/codemods/emotion-to-compiled/__tests__/emotion-to-compiled.test.tsx:328]: the imports that aren't supported are nice - good job with the comments!

**Review by madou — APPROVED** [review]
lgtm

**Pull request merged**